homeassistant/components/modbus/modbus.py

```diff
@@ -237,8 +236,9 @@ def read_coils(self, unit, address, count):
                 result = self._client.read_coils(address, count, **kwargs)
             except ModbusException as exception_error:
                 self._log_error(exception_error)
-                return None
-            if isinstance(result, (ExceptionResponse, IllegalFunctionRequest)):
+                result = exception_error
+                pass
```

Comment:
`pass` does nothing here.
```suggestion
```

Reply:
You need to look at the file and not only the diff
```yaml
except ModbusException as exception_error:
                self._log_error(exception_error)
                result = exception_error
                pass
if not hasattr(result, "registers"):
                self._log_error(result)
                return None
```

It instructs the exception to continue with the next statement, and it also a good advice when you read the code.

Reply:
But I see that pylint also complains, so away it goes.